src/specialfunctions.jl

```diff
@@ -61,6 +61,14 @@ function pochhammer(x::Number,n::UnitRange{T}) where T<:Real
     ret
 end
 
+function ogamma(x::Number)
+    if isinteger(x) && x<0
+        0.0
```

Comment:
This should be type stable: `zero(float(x))`

Reply:
Also, why not `x ≤ 0`?

Reply:
Because the original `gamma` already returns `Inf`
`gamma(0.0)=Inf`
`gamma(-0.0)=-Inf`